File: app/src/main/java/ir/arapp/arappmain/Util/Adapters/SingleDirectionRecyclerView.java
package ir.arapp.arappmain.Util.Adapters;

import android.content.Context;
import android.util.AttributeSet;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.recyclerview.widget.RecyclerView;

import org.jetbrains.annotations.NotNull;

public class SingleDirectionRecyclerView extends RecyclerView {
    SingleScrollDirectionEnforcer enforcer = new SingleScrollDirectionEnforcer();
    public SingleDirectionRecyclerView(@NonNull @NotNull Context context) {
        super(context);
        init();
    }

    public SingleDirectionRecyclerView(@NonNull @NotNull Context context, @Nullable @org.jetbrains.annotations.Nullable AttributeSet attrs) {
        super(context, attrs);
        init();
    }

    public SingleDirectionRecyclerView(@NonNull @NotNull Context context, @Nullable @org.jetbrains.annotations.Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        init();
    }

    void init(){
        addOnScrollListener(enforcer);
        addOnItemTouchListener(enforcer);
    }
}

File: app/src/main/java/ir/arapp/arappmain/view/utils/SingleDirectionRecyclerView.kt
package ir.arapp.arappmain.view.utils

import android.content.Context
import android.util.AttributeSet
import androidx.recyclerview.widget.RecyclerView

class SingleDirectionRecyclerView : RecyclerView {
    var enforcer = SingleScrollDirectionEnforcer()

    constructor(context: Context) : super(context) {
        init()
    }

    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs) {
        init()
    }

    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(
        context,
        attrs,
        defStyleAttr
    ) {
        init()
    }

    fun init() {
        addOnScrollListener(enforcer)
        addOnItemTouchListener(enforcer)
    }
}